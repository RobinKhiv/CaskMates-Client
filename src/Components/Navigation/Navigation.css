*{ 
    box-sizing: border-box;
  }
/* li{
    list-style: none;
} */
nav *{
    display:inline-block;
  }
.Header{
  height: 4px;
}
nav{
    margin-left: auto;
    margin-right: auto;
    background-color: rgb(94, 65, 54);
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    height: 24px;
}
.listNav{
  background-color: rgb(94, 65, 54);
  border: rgb(94, 65, 54);
}
.nav-a, .brand, .listNav{
  font-size: 13px;
    font-weight: 400;
    color: #aaaaaa;
    font-family: 'Oswald', sans-serif;
    font-weight: 600;
    text-decoration: none;
    letter-spacing: 1px;
    font-weight: 700;
    
    
    /* transform: scaleX(1.4); */
    /* transform-origin: 0 0; */
    padding-left: 2px;
  }
  img{
    mix-blend-mode: multiply;
  }
  /* ul {
    padding-left: 0;
  } */
  .nav-a {
    margin: 5px;
    margin-left: 5px;
  }
  .navbar-anchors{ 
    float: right;
    text-decoration: none;
    /* position: absolute; */
    margin-right: 5px;
  }