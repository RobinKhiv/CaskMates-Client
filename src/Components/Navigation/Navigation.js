import React, { Component } from 'react';
import {Link} from 'react-router-dom';
import TokenService from '../../Services/token-api-service';
import WhiskeyContext from '../../Context/WhiskeyContext';
import './Navigation.css';

export default class Navigation extends Component {
  static contextType = WhiskeyContext;

  handleLogoutClick = () => {
    TokenService.clearAuthToken()
    this.context.userLogout();
  }

  renderLogoutLink() {
    return (
      <div className='Header__logged-in'>
        <Link className='nav-a logout' onClick={this.handleLogoutClick} to='/'>
          <img alt='logout icon' height='14' src='https://cdn2.iconfinder.com/data/icons/picons-essentials/57/logout-512.png'/>
        </Link>
      </div>
    )
  }

  renderLoginLink() {
    return (
        <div className='Header__not-logged-in'>
          <Link className='nav-a log'
            to='/login'><img alt="login icon" src='https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSrdZKzkouFPEorhZMdksNlgtVyANA_0YJl1NYGqlpaELG4TslJig' height='14'></img>
          </Link>
        </div>
      )
  }

  renderUserFeaturesLink() {
    return (
      <React.Fragment>
        <Link className='nav-a' to='/addWhiskey'>Add Whiskey</Link>
        <span>|</span>
        <Link className='nav-a' to='/whiskeyList'>My List</Link>
        <span>|</span>
      </React.Fragment>
    )
  }
  
  render() {
    return (
      <nav role='navigation' className='navigation'>
        <Link to='/' className='brand'>Caskmates</Link>
        <div className='navbar-anchors'>
          <Link className='nav-a' to='/whiskeys'>Whiskey Lookup</Link>
          <span>|</span>
          {TokenService.hasAuthToken() 
          ? this.renderUserFeaturesLink() 
          : ''}
          {TokenService.hasAuthToken()
          ? this.renderLogoutLink()
          : this.renderLoginLink()}
        </div> 
      </nav>
    )
  }
}
