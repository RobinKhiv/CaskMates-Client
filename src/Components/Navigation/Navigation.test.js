import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import Navigation from './Navigation';

describe('<Navigation />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <Navigation />
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

  it('renders without crashing', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Navigation />
        </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot(); 
  });
});