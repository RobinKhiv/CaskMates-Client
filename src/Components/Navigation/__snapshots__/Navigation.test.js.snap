// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> renders without crashing 1`] = `
<nav
  className="navigation"
  role="navigation"
>
  <a
    className="brand"
    href="/"
    onClick={[Function]}
  >
    Caskmates
  </a>
  <div
    className="navbar-anchors"
  >
    <a
      className="nav-a"
      href="/whiskeys"
      onClick={[Function]}
    >
      Whiskey Lookup
    </a>
    <span>
      |
    </span>
    
    <div
      className="Header__not-logged-in"
    >
      <a
        className="nav-a log"
        href="/login"
        onClick={[Function]}
      >
        <img
          alt="login icon"
          height="14"
          src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSrdZKzkouFPEorhZMdksNlgtVyANA_0YJl1NYGqlpaELG4TslJig"
        />
      </a>
    </div>
  </div>
</nav>
`;
